Floofbot/Services/EventHandlerService.cs

```diff
@@ -114,6 +115,28 @@ private async Task CheckUserAutoban(IGuildUser user)
                 }
             }
         }
+
+        // rfurry rules gate
+        public async Task RulesGate(SocketMessage msg)
+        {
+            var userMsg = msg as SocketUserMessage;
+            if (msg == null || msg.Author.IsBot)
+                return;
+
+            // rules gate info
+            ulong rulesChannelId = 225980129799700481;
+            ulong rulesServerId = 225980129799700481;
+            ulong readRulesRoleId = 494149550622375936;
+            string rulesBypassString = "yes";
+
+            if (msg.Channel.Id == rulesChannelId && userMsg.Content.ToLower().Contains(rulesBypassString)) 
+            {
+                var user = (IGuildUser)msg.Author;
+                SocketGuild guild = _client.GetGuild(rulesServerId);
```

Comment:
Is using `GetGuild` necessary? Can you use `user.Guild` (and cast to `SocketGuild` if necessary) instead? The assumption is that if the channel ID matches the defined rules channel, the message already originates from the target server.

Reply:
changed